fixes to Dockerfile and test data

realmarcin opened the pull request.

realmarcin committed:
many changes:

- switched R install to conda
- this required installing conda
- most R packages installed through conda but only after enabling addition conda channels
- one R package only worked via standard R install

realmarcin committed:
replaced DvH and Rhodobacter test data with new uploaded/downloaded RefSeq genomes matching latest spec on ci.kbase.us

realmarcin committed:
updated feature count in test

realmarcin committed:
updated genome file

realmarcin committed:
new assembly file

realmarcin committed:
fixes and updates to tests

realmarcin committed:
cleaned up Dockerfile

upped version

one more test assembly

realmarcin committed:
removed genbank_handle_ref from DvH Genome json

realmarcin committed:
Revert "removed genbank_handle_ref from DvH Genome json"

This reverts commit d537ca7ba3d8768341d6dfe621299269dd003ad6.

realmarcin committed:
removed assembly_ref from Rhodobacter Genome json

realmarcin committed:
removing problematic handles

realmarcin committed:
removing problematic handles

realmarcin committed:
removed dependency on Assembly for Rhodobacter test

realmarcin committed:
fixed class declaration after Assembly file load purge

realmarcin committed:
removing one more assembly ref and purging from test

realmarcin committed:
purging Assembly and refs

MrCreosote commented on test/src/us/kbase/kbasefeaturevalues/test/KBaseFeatureValuesServerTest.java:
```diff
@@ -126,7 +126,7 @@ public static void init() throws Exception {
         Map<String, Object> contigsetData = new LinkedHashMap<String, Object>();
         contigsetData.put("contigs", new ArrayList<Object>());
-        contigsetData.put("id", "1945.contigset");
+        contigsetData.put("id", "1945.assembly");
```
Adding an assembly test is good, but the contig set test should be left in place to test for regressions on handling contig sets, which still exist in the system.

realmarcin replied:
Well, the Assembly and ContigSet components are removed because they complicated the tests and none of that data is actually ever used in these apps. These apps just need the Genome object to get a list of feature ids. So the tests run on somewhat slimified Genome objects though slimming only applies to handles and Assembly/ContigSet.

realmarcin replied:
Except for this one test with DvH, this one still uses ContigSet/Assembly. I can purge it as needed but tests pass for me and Zhenyuan.

MrCreosote replied:
If the app never touches the dependency objects, is it possible to save a genome without either the assembly or contigset? If so do it that way. Having the CS or Assembly as part of the tests will make people think they're actually tested and important.

MrCreosote reviewed (changes requested):
I can't really offer useful feedback on the dockerfile changes as I'm not particularly familiar with conda or R.

MrCreosote commented:
Also I don't understand the FeatureValues module at all

realmarcin committed:
removed ContigSet depdendency for DvH test

realmarcin committed:
- checking in .gbff source files in case test .json files need upgrades
- explained test data prep in README (for case of backwards incompatible Genome object changes)

MrCreosote commented on README.md:
```diff
@@ -9,6 +9,10 @@ of expression values organized as two-dimensional matrix and assigned to
 genome features on one dimension and to different sorts conditions on second 
 dimension.
 
+Notes for preparing test data
```
This doesn't explain how to generate the json files from the gbff files AFAICT

realmarcin committed:
more test data creation details

realmarcin committed:
purging last assembly and contigset files

realmarcin reviewed (commented):
These modules only require the feature id data from the Genome object hence for testing purposes I purged all Assembly and ContigSet objects as well as their references. Using the slightly slimmer Genome objects simplifies and isolates the tests.

MrCreosote commented:
Hmm, can't comment in your thread for some reason.

> Except for this one test with DvH, this one still uses ContigSet/Assembly. I can purge it as needed but tests pass for me and Zhenyuan.

Purge it, makes things easier to understand for the next person who comes along and has to work on this module.

Tianhao-Gu reviewed (changes requested):
I am not too useful here since I never worked on FeatureValues. But @realmarcin asked me to review. It seems like docker won't build on master branch. And docker builds successfully with this PR. There is also no impl code changes. So I don't know how this will fix anything. Maybe the imported library changed?

More importantly, local tests failed:
    [junit] Running us.kbase.clusterservice.test.ClusterServiceTest
    [junit] Testsuite: us.kbase.clusterservice.test.ClusterServiceTest
    [junit] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 14.46 sec
    [junit] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 14.46 sec
    [junit] 
    [junit] Testcase: pyTest took 3.542 sec
    [junit] Testcase: rEstimateKNewTest took 1.8 sec
    [junit] Testcase: rTest took 9.068 sec
    [junit] Running us.kbase.kbasefeaturevalues.test.KBaseFeatureValuesServerTest
    [junit] Testsuite: us.kbase.kbasefeaturevalues.test.KBaseFeatureValuesServerTest
    [junit] Tests run: 0, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 83.14 sec
    [junit] Tests run: 0, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 83.14 sec
    [junit] ------------- Standard Output ---------------
    [junit] Test workspace [test_KBaseFeatureValues_1581718881765] was deleted
    [junit] ------------- ---------------- ---------------
    [junit] 
    [junit] Testcase: us.kbase.kbasefeaturevalues.test.KBaseFeatureValuesServerTest took 0 sec
    [junit] 	Caused an ERROR
    [junit] Object #1, Desulfovibrio_vulgaris_Hildenborough.genome failed type checking:
    [junit] Object doesn't have required fields : [taxon_ref], at 
    [junit] us.kbase.common.service.ServerException: Object #1, Desulfovibrio_vulgaris_Hildenborough.genome failed type checking:
    [junit] Object doesn't have required fields : [taxon_ref], at 
    [junit] 	at us.kbase.common.service.JsonClientCaller.jsonrpcCallStatic(JsonClientCaller.java:411)
    [junit] 	at us.kbase.common.service.JsonClientCaller.jsonrpcCall(JsonClientCaller.java:329)
    [junit] 	at us.kbase.common.service.JsonClientCaller.jsonrpcCall(JsonClientCaller.java:298)
    [junit] 	at us.kbase.workspace.WorkspaceClient.saveObjects(WorkspaceClient.java:487)
    [junit] 	at us.kbase.kbasefeaturevalues.test.KBaseFeatureValuesServerTest.init(KBaseFeatureValuesServerTest.java:129)
    [junit] 

BUILD FAILED
/kb/module/build.xml:109: Test us.kbase.kbasefeaturevalues.test.KBaseFeatureValuesServerTest failed

Total time: 1 minute 39 seconds
make: *** [Makefile:80: test] Error 1
Shutting down callback server...

landml commented:
@realmarcin  and @MrCreosote 
There is a new ticket reporting problems with these apps - https://kbase-jira.atlassian.net/browse/PUBLIC-974. Any attention to this problem will be appreciated.

realmarcin commented:
the notes for creating test data json files are in the repo README.md

https://github.com/kbaseapps/FeatureValues/tree/marcin_fixes

MrCreosote commented on test/data/upload8/NewFakeData2.3.json:
```diff
@@ -1 +1 @@
-{"data":{"col_ids":["SomeCondition","OtherCondition"],"row_ids":["RSP_0049","RSP_0046","RSP_2436","RSP_2435","RSP_1584","RSP_2229","RSP_1588"],"values":[[0.1,0.5],[-1.8,-0.2],[5.0,1.2],[0.4,0.8],[-1.2,0.6],[0.9,-0.2],[0.7,-1.0]]},"feature_mapping":{"RSP_0046":"RSP_0046","RSP_0049":"RSP_0049","RSP_1584":"RSP_1584","RSP_1588":"RSP_1588","RSP_2229":"RSP_2229","RSP_2435":"RSP_2435","RSP_2436":"RSP_24 ...
\ No newline at end of file
+{"data":{"col_ids":["SomeCondition","OtherCondition"],"row_ids":["RSP_1428","RSP_1430","RSP_1432","RSP_1433","RSP_1434","RSP_1435","RSP_1436"],"values":[[0.1,0.5],[-1.8,-0.2],[5.0,1.2],[0.4,0.8],[-1.2,0.6],[0.9,-0.2],[0.7,-1.0]]},"feature_mapping":{"RSP_0046":"RSP_0046","RSP_0049":"RSP_0049","RSP_1584":"RSP_1584","RSP_1588":"RSP_1588","RSP_2229":"RSP_2229","RSP_2435":"RSP_2435","RSP_2436":"RSP_24 ...
```
Out of curiosity why change these IDs?

MrCreosote commented:
Are there any differences between this and the `marcin_fixes` branch?

realmarcin committed:
reverting test ids

realmarcin committed:
purged last assembly_ref

MrCreosote commented:
per @realmarcin the `marcin_fixes` branch is identical to this PR other than the 2 most recent commits

realmarcin committed:
intermedite Dockerfile with COPY issue

Commit:
requirements.txt for condawq

Commit:
working Dockerfile, but old sdk

MrCreosote commented on Dockerfile:
```diff
@@ -16,32 +21,64 @@ RUN apt-get update && apt-get install -y \
 COPY ./ /kb/module
 RUN mkdir -p /kb/module/work
-RUN chmod -R a+rw /kb/module
+RUN chmod 777 /kb/module
```
You definitely want the recursive chmod, this is known to cause weird issues if you don't do it

MrCreosote replied:
All the other dockerfiles use the a+rw syntax, so unless there's a good reason to use 777 here I'd say switch it back

MrCreosote commented on requirements.txt:
```diff
@@ -0,0 +1,7 @@
+r
```
I don't think this is used anywhere?

MrCreosote commented on kbase.yml:
```diff
@@ -11,10 +11,10 @@ service-language:
 
 owners:
-    [rsutormin, msneddon, psnovichkov, marcin, srividya22]
+    [marcin, rsutormin, msneddon, psnovichkov, srividya22]
```
Remove everyone except yourself IMO, they don't work here anymore

MrCreosote reviewed (changes requested):
Def want to fix the chmod, the other two issues are nice to haves

Commit:
reverted chmod
removed requirements.txt
updated owners

MrCreosote commented:
Once you tell me tests pass I'll merge

Commit:
reverting subset of gene ids

Commit:
module builds
tests pass on ci

Commit:
cleanup

Commit:
whitespace

MrCreosote commented:
Per @realmarcin tests pass, so merging.
